pkg/scheduling/volumeusage.go

```diff
@@ -130,22 +123,24 @@ func resolveDriver(ctx context.Context, kubeClient client.Client, volumeName str
 			return driverName, nil
 		}
 	}
-	if storageClassName != "" {
-		driverName, err := driverFromSC(ctx, kubeClient, storageClassName)
-		if err != nil {
-			return "", err
-		}
-		if driverName != "" {
-			return driverName, nil
-		}
+
+	driverName, err := driverFromSC(ctx, kubeClient, storageClassName)
+	if storageClassName == "" || errors.IsNotFound(err) {
+		logging.FromContext(ctx).Warnf("failed to update volume limits for volume with unbound PVC, storage class not found \"%s\"", storageClassName)
```

Comment:
We should avoid warnings at all costs. In general, it's either an error or an info/debug. If we have a storage class that doesn't exist and doesn't have a bound volume, this should be an error

Reply:
Got it, I figured since we could still progress a warning may have been more appropriate but I'm happy to change it to an error.